surefire-provider/src/test/java/org/arquillian/smart/testing/surefire/provider/SmartTestingProviderTest.java

```diff
@@ -123,12 +135,67 @@ public void test_when_invoke_is_called_with_whole_set_of_classes() throws Except
             iterable -> iterableContains(iterable, expectedClassesToRun)));
     }
 
+    @Test
+    public void should_add_test_method_selection_to_test_list_resolver() throws TestSetFailedException, InvocationTargetException {
+        // given
+        TestSelection strategy = new TestSelection(ATest.class.getName(), "strategy");
+        strategy.getTestMethodNames().add("firstMethod");
+        strategy.getTestMethodNames().add("secondMethod");
+
+        prepareTestListResolver();
+
+        SmartTestingInvoker smartTestingInvoker = prepareSTInvoker(Arrays.asList(strategy));
+        SmartTestingSurefireProvider provider =
+            new SmartTestingSurefireProvider(providerParameters, providerFactory, smartTestingInvoker);
+
+        // when
+        provider.invoke(null);
+
+        // then
+        softly.assertThat(testRequest.getTestListResolver())
+            .hasMethodPatterns(true)
+            .hasIncludedMethodPatterns(true)
+            .hasExcludedMethodPatterns(true)
+            .includedPatterns()
```

Comment:
How about renaming `includedPatterns()` to something like `hasIncludedPatterns ()` ?

Reply:
`includedPatterns()` doesn't verify anything, it just returns an instance of an asserstion for the collection: `Assertions.assertThat(actual.getExcludedPatterns());`